src/selection.rs

```diff
@@ -0,0 +1,112 @@
+use crate::utils::{dist_between_points, Point};
+use std::time::Instant;
+
+type Milliseconds = u128;
+
+const CLICK_TOLERANCE: Milliseconds = 300;
```

Comment:
```suggestion
const CLICK_TOLERANCE: Duration = Duration::from_millis(300);
```

Might as well use a duration here since we can (still requires more changes to anywhere `CLICK_TOLERANCE` gets used though)